### ISSUE TITLE
Allow Ia5String to be created from String.

### ISSUE THREAD

**Issue by turbocool3r** [RustCrypto/formats#1057]
Currently `Ia5String` constructor looks like this:

```rust
pub fn new<T>(input: &T) -> Result<Self>
where T: AsRef<[u8]> + ?Sized,
{
    ...
}
```

This is different from the `OctetString` constructor which looks like this:

```rust
pub fn new(bytes: impl Into<Vec<u8>>) -> Result<Self> {
    ...
}
```

The former doesn't allow to create an `Ia5String` from a `String` without additional allocations. I suggest changing it to something like the `OctetString` constructor or at least implementing `TryFrom<String>`.

**Comment by tarcieri** [comment]
Yes, the APIs of the various string types should be more consistent

**Comment by turbocool3r** [comment]
@tarcieri will changing the `input` argument type to `impl Into<String>` or `impl Into<Vec<u8>>` be considered a breaking change that can't be accepted currently?

**Comment by tarcieri** [comment]
Technically yes, but I imagine in practice it would probably be OK.

FWIW I sketched out a unifying solution for all of the string types yesterday, but that would definitely be a breaking change: #1061

**Comment by tarcieri** [comment]
Also as a non-breaking change we can add `TryFrom<String> for Ia5String`

**Comment by turbocool3r** [comment]
> Also as a non-breaking change we can add `TryFrom<String> for Ia5String`

Implemented this in #1064.
